packages/router/src/Set.tsx

```diff
@@ -65,7 +65,11 @@ export function Set<WrapperProps>(props: SetProps<WrapperProps>) {
 
   if (privateSet && unauthorized()) {
     if (loading) {
-      return whileLoadingAuth?.() || null
+      if (React.isValidElement(WhileLoadingAuth)) {
```

Comment:
Should we throw an error if it isn't a valid react component (in dev) instead of just returning `null`?

Reply:
Update: `React.isValidElement(WhileLoadingAuth)` returns `false`, so there's something I'm still missing...